### ISSUE TITLE
New items not saving, but editing works

### ISSUE THREAD

**Issue by philip** [frapi/frapi#8]
This could be a setup problem, although I'm unsure how to debug it (yet). Note: I only followed the guide. Hopefully it's something simple.

I set up frapi, and it seemingly works except added new items aren't being saved. For example, I'm unable to add new actions/errors/partners. However, editing the current ones works. This also applies to configurations where editing works, but adding new items does not. 

I now see that adding a partner simply emits a near blank page with "bool(false)" so that should be easier to debug, whereas the other items act as the though the item was saved (e.g., "Error code E_FOO added.")

File permissions seem fine, and editing the XML files manually works too. And "Sync Now" was used.  Also, the docs don't mention CouchDB setup, or where/if it's used yet.

**Comment by davidcoallier** [comment]
1) There's no CouchDB setup bundled with FRAPI. There's only the model file in custom models and this is only used to connect to CouchDB if you do have CouchDB :) As we encourage and love Couch we bundled the lib however there's no Couch setup or anything, it's purely for example and personal usage purposes.

2) Regarding the other issues about not being able to add new items I'm a bit confused. I've never seen FRAPI do that apart from bad permissions on OSX. 

Could you show me the permissions that are set on both 

src/frapi/custom 
and the
src/frapi/admin/application/config/app

directories. If the XML files are getting modified then this is odd :)

Could you also give me infos on  your setup? Distro, php version, etc?

Cheers

**Comment by philip** [comment]
In an effort to debug this, I went a little crazy with chmod -R 777 ... so:

drwxrwxrwx 5 www-data www-data 4096 May 22 00:12 custom
drwxrwxrwx 2 www-data www-data 4096 May 22 00:12 custom/Action
drwxrwxrwx 2 www-data www-data 4096 May 22 00:12 config/app

And yes, the XML files are getting updated for edits, but not additions. The associated PHP files too. I'm close to losing my mind, but my sysadmin skills leave a lot to be desired. Is there a 'debug all' switch of some kind? The 'partners/add' returning bool(false) whereas the other */add don't (yet don't touch XML either) seems like an indicator of something (and/or a bug of its own). Also, the stats/ page says "Sorry! A system error prevented your action from being executed" might also lead to the problem.

Setup: NGinx 0.7.63 + PHP-FPM (5.3.2) + Ubuntu 9.10 and www-data runs worker processes for both nginx and php-fpm.

**Comment by davidcoallier** [comment]
Odd, it does sound as a bug indeed, want to jump on #frapi on freenode?

**Comment by philip** [comment]
Sure, I'm there now.

**Comment by davidcoallier** [comment]
Ok so do you have any error logs for PHP that could indicate something is going wrong? If not I'll go on and setup the same server that you have and see if I can reproduce :)

**Comment by davidcoallier** [comment]
ok I found it. sorry about this, last minute commit before the  release :))

**Comment by philip** [comment]
:)

**Comment by arski** [comment]
Hi guys, I'm getting the exact same issue and at least I know what I did - I tried to add parameters to an action by directly editing the actions.xml file (there were a lot and I didn't want to add them manually..).. After doing this, the interface says that it's saving stuff properly when editing (when it's actually not) and also giving the same error as described above when trying to delete/add an action.

I tried restoring actions.xml to the old state, but that didn't do the trick.. and clearing Actions.enabled-private cache didn't help either. Any idea how to fix this?

PS. Looking at the above commit, it doesn't look like anything was actually changed? says 0 modified files to me.

Thanks!
